Makes teleporters use more power

Detritus-Ha opened the pull request:
:cl:
tweak: Increased teleporter power usage.
tweak: Teleporters disengage automatically after a while.
/:cl:
Teleporters currently use 4000-9000W while active. That's absolutely ridiculous for a machine that creates a safe and stable mini-wormhole.

Detritus-Ha committed:
Update teleporter.dm

Karlious commented:
So uh, what did you change it too? :D

Detritus-Ha commented:
> 
> 
> So uh, what did you change it too? :D

Teleporter activation: 5kW -> 200kW
Teleporter power usage while active: 4kW -> 200kW
Teleporter use: 5kW -> 100kW

SierraKomodo commented:
In testing, new values seem too high. An APC without connection to a SMES immediately dies if the teleporter is on, and the power draw on the networked SMES increases a whopping 200kw if a teleporter is just left on - Which it will be if a teleporter is used, because there's currently no way to turn it off after using one unless you had someone come with you solely to turn it off then walk.

Detritus-Ha commented:
>In testing, new values seem too high. An APC without connection to a SMES immediately dies if the teleporter is on, and the power draw on the networked SMES increases a whopping 200kw if a teleporter is just left on.

The teleporter stays on for long enough to send several people through with just a charged APC, I would actually increase the teleporter power usage to 0.5-1MW if substations were not so weak by default.

Ilysen commented:
I've talked about this in the #pr-feedback channel on Discord, but you don't seem to be in the server, so I'll mirror my own thoughts here. I'm uncertain what problem this sets out to address, and I'm more uncertain that this is a beneficial change to address it. Teleporters are very commonly used by both engineering and medical, and while the ship *can* support the power generation to handle them, these changes mean that teleporters will draw huge amounts of power while active, with no way to disable them without having someone else turn it off for you once you go through it.

If I may ask, what's the problem this PR is intended to address? The description posits that it's ridiculous, but I'm uncertain if this is from a balance standpoint or a realism standpoint. Engineering and medical staff both rely heavily on teleporters - especially engineering, because moving portable atmospherics devices across decks without a teleporter is hellish.

Detritus-Ha commented:
>If I may ask, what's the problem this PR is intended to address? The description posits that the current draw is "ridiculous", but I'm uncertain if this is from a balance standpoint or a realism standpoint. 

Kind of both. Teleporters are extremely powerful tools and are _at least in my opinion_ something crew should not constantly rely on. I mostly play engineering and I think the teleporters are way too handy, and completely ruin many great moments. 
For a long time it was a case of _"this could use work, but I am not sure what exactly should be done"_ , but a few days ago someone pointed out that the teleporters are basically free during the circuit teleporter discussion.
While increasing the needed power does not "fix" teleporters it does at least make them less magic and more machine.

If you really do think 200kW power usage is high for a machine that basically bends space, it can be easily reduced.

Ilysen commented:
I think that the amount is very high without further changes. A good idea that came up in the Discord channel was allowing you to set the teleporter to auto-close after a few seconds or certain amount of uses in order to counteract the huge drain now, which I think is a good idea, and certainly feasible to do with just HTML screens. I would refer to body scanner code for guidance on how you might do something like that, as it features very extensive use of data and interaction buttons.

SierraKomodo commented:
Here's something to consider - What effect is increasing the power usage of the teleporters themselves going to have on a round? Essentially nothing. Engineering already on a regular will run an engine setup that produces so much more energy than is needed that this change won't actually affect anything, beyond killing a room's power if it loses connection to substations for whatever reason.

The issue brought up of _portable_ teleporters isn't going to be fixed or adjusted at all by this.

Detritus-Ha committed:
Teleporters disengage automatically

-Teleporters disengage automatically (after 20 seconds by default)
-Added new verb for adjusting automatic teleporter shutdown

Detritus-Ha commented:
>I think that the amount is very high without further changes. A good idea that came up in the Discord channel was allowing you to set the teleporter to auto-close after a few seconds or certain amount of uses in order to counteract the huge drain now

Easily done. The result is a bit rough but it seems to work very nicely with a verb on the projector to set the timer.

>What effect is increasing the power usage of the teleporters themselves going to have on a round? Essentially nothing.

Exactly. If I wanted to make a change that has a big effect on the round...I would PR teleport cooldown or something similar, but I am sure everyone would hate absolutely hate that.

Spookerton commented:
Although I approve of the sentiment, I don't think I want to consider such a large cost without making them wired machines instead, simply for mechanical raisins.

Detritus-Ha commented:
> I don't think I want to consider such a large cost without making them wired machines instead, simply for mechanical raisins.

I _really_ should have thought about that, I think I will just close this and maybe redo the whole thing later.
